### PR TITLE
Fix Pkg Build Order to Mirror docker-compose.yml

### PR DESCRIPTION
**Pull request by shamrickus**
This changes `pkg` to (by default) build services in the order listed in `infrastructure/docker/build/docker-compose.yml` as opposed to alphabetical order. 

This issue was originally fixed in #5368, but was reintroduced in #6274 as `docker-compose config --services` lists services in alphabetical order.
<hr/>

## Which Traffic Control components are affected by this PR?

- Build

## What is the best way to verify this PR?
Run `./pkg source` and then extract the produced source and run `./pkg -v` again. Weasel should not fail from build artifacts such as [these](https://lists.apache.org/thread/okndcgcxn3qr5nkqg1tokbf0w6p38h9n)

## If this is a bugfix, which Traffic Control versions contained the bug?

 - 6.1.0
 - 7.0.0 RC0
 - master

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by zrhoffman** [comment]
> but was reintroduced in #6274 as `docker-compose config --services` lists services in alphabetical order.

Upon closer inspection, `pkg` has used

```shell
docker-compose config --services
```

to list services since its creation. The Go implementation of `docker-compose config --services` alphabetizes the services, while the (now-unsupported) Python version,

```shell
python -m compose config --services
```
does not.

File: pkg [hunk 1]
```diff
@@ -155,7 +155,7 @@ while getopts h78abdf:lopqvsSL opt; do
 	esac
 done
 
-PROJECTS="$("${COMPOSECMD[@]}" -f $COMPOSE_FILE config --services)"
+PROJECTS="$(cat infrastructure/docker/build/docker-compose.yml | grep -E '^\s\s[a-zA-Z_-]+:$' | sed 's/[: ]//g')"
```
Comment (zrhoffman) [inline pkg:158]: Parsing YAML with `grep` and `sed` does not seem so maintainable. What are your thoughts about rearranging the output of `docker-compose config --services` to have `weasel` first?

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: pkg (hunk 1 above)
Comment (shamrickus) [reply pkg:158]: That sounds like a better solution to me

File: pkg [hunk 2]
```diff
@@ -155,7 +155,7 @@ while getopts h78abdf:lopqvsSL opt; do
 	esac
 done
 
-PROJECTS="$(cat infrastructure/docker/build/docker-compose.yml | grep -E '^\s\s[a-zA-Z_-]+:$' | sed 's/[: ]//g')"
+PROJECTS="weasel"$'\n'"$("${COMPOSECMD[@]}" -f $COMPOSE_FILE config --services | sed '/^weasel$/d')"
```
Comment (zrhoffman) [inline pkg:158]: This looks good, but would you mind including a comment that explains why it's like that?

**Review by zrhoffman — APPROVED** [review]
Looks good!

**Pull request merged**